### PR TITLE
add hmacMd5 to StringUtil

### PR DESCRIPTION
**Pull request by goszjacek**

### CODE REVIEW

File: scex-util/src/main/scala/com/avsystem/scex/util/function/StringUtil.java [hunk 1]
```diff
@@ -48,4 +48,6 @@
     String replace(String str, String find, String replacement);
 
     String join(Collection<String> list, String separator);
+
+    String hmacMd5(String str, String key);
```
Comment (ghik) [inline scex-util/src/main/scala/com/avsystem/scex/util/function/StringUtil.java:52]: Cosmetic stuff but `Md5` looks somewhat weird... `MD5`?

Comment (goszjacek) [reply scex-util/src/main/scala/com/avsystem/scex/util/function/StringUtil.java:52]: ok, I'll change it

Comment (ghik) [reply scex-util/src/main/scala/com/avsystem/scex/util/function/StringUtil.java:52]: I think you may also want to add this method into `StringMiscOps` (extensions methods for `String` type). The `StringUtil` interface and implementation is somewhat legacy as it comes from very old days of SpEL (Spring Expression Language).

Comment (goszjacek) [reply scex-util/src/main/scala/com/avsystem/scex/util/function/StringUtil.java]: thank you for the suggestion, I added it

Comment (goszjacek) [reply scex-util/src/main/scala/com/avsystem/scex/util/function/StringUtil.java]: oh, I see I need to run the tests

File: scex-util/src/main/scala/com/avsystem/scex/util/StringMiscOps.scala [hunk 2]
```diff
@@ -113,6 +113,10 @@ final class StringMiscOps(private val wrapped: String) extends AnyVal {
     "returns the result as a 32 character hex string.")
   def md5Hex: String = DigestUtils.md5Hex(wrapped.getBytes(StandardCharsets.UTF_8))
 
+  @Documentation("Calculates HMAC MD5 digest from contents of this string and " +
+    "returns the result as a 32 character hex string.")
+  def hmacMD5(str: String, key: String): String = new HmacUtils(HmacAlgorithms.HMAC_MD5, key).hmacHex(str)
```
Comment (ghik) [inline scex-util/src/main/scala/com/avsystem/scex/util/StringMiscOps.scala:118]: This is an extension method - `wrapped` is the string being hashed.

**Review by ghik — CHANGES_REQUESTED** [review]

**Review by ghik — APPROVED** [review]

**Pull request merged**